src/main/webapp/js/playGame.js

```diff
@@ -23,8 +17,24 @@ async function initMapToPlayGame() {
       window.location.replace('index.html');
       return;
     }
-    var initStage =  await getStage(data.stages[0]);
-    var startingLocation = {lat: initStage.startingLocation.latitude, lng: initStage.startingLocation.longitude};
+    var userProgress = getUserProgress();
+    var startingLocation;
+    var initStage;
+    var stageID;
+    console.log(userProgress);
+    if (userProgress == null) {
+      stageID = data.stages[0];
+      initStage =  await getStage(stageID);
+      startingLocation = {lat: initStage.startingLocation.latitude, lng: initStage.startingLocation.longitude};
+    } else {
+      console.log('from progress');
+      stageID = userProgress.stageID;
+      initStage =  await getStage(stageID);
+      startingLocation = {lat: userProgress.location.latitude, lng: userProgress.location.latitude.longitude};
+      console.log(startingLocation);
+
```

Comment:
delete empty line

Reply:
That is already removed in my current version, but yes thank yoi.